src/prefect/cli/cloud.py

```diff
@@ -35,110 +49,234 @@
-    and allow them to select one.
+    Given a list of options, display the values to user in a table and prompt them
+    to select one.
 
     Args:
-        workspaces: List of workspaces to choose from
+        options: A list of options to present to the user.
+            A list of tuples can be passed as key value pairs. If a value is chosen, the
+            key will be returned.
 
     Returns:
-        str: the selected workspace
+        str: the selected option
     """
 
-    workspaces = sorted(workspaces)
     current_idx = 0
-    selected_workspace = None
+    selected_option = None
+
+    def build_table() -> Table:
+        """
+        Generate a table of options. The `current_idx` will be highlighted.
+        """
+
+        table = Table(box=False, header_style=None, padding=(0, 0))
+        table.add_column(
+            f"? [bold]{prompt}[/] [bright_blue][Use arrows to move; enter to select]",
+            justify="left",
+            no_wrap=True,
+        )
 
-    with Live(
-        build_table(current_idx, workspaces), auto_refresh=False, console=app.console
-    ) as live:
-        while selected_workspace is None:
+        for i, option in enumerate(options):
+            if isinstance(option, tuple):
+                option = option[1]
+
+            if i == current_idx:
+                # Use blue for selected options
+                table.add_row("[bold][blue]> " + option)
+            else:
+                table.add_row("  " + option)
+        return table
+
+    with Live(build_table(), auto_refresh=False, console=console) as live:
+        while selected_option is None:
             key = readchar.readkey()
 
             if key == readchar.key.UP:
                 current_idx = current_idx - 1
                 # wrap to bottom if at the top
                 if current_idx < 0:
-                    current_idx = len(workspaces) - 1
+                    current_idx = len(options) - 1
             elif key == readchar.key.DOWN:
                 current_idx = current_idx + 1
                 # wrap to top if at the bottom
-                if current_idx >= len(workspaces):
+                if current_idx >= len(options):
                     current_idx = 0
             elif key == readchar.key.CTRL_C:
                 # gracefully exit with no message
                 exit_with_error("")
             elif key == readchar.key.ENTER:
-                selected_workspace = workspaces[current_idx]
+                selected_option = options[current_idx]
+                if isinstance(selected_option, tuple):
+                    selected_option = selected_option[0]
+
+            live.update(build_table(), refresh=True)
+
+        return selected_option
+
+
+async def login_with_browser() -> str:
+    """
+    Perform login using the browser.
+
+    On failure, this function will exit the process.
+    On success, it will return an API key.
+    """
+    # TODO: Search for a valid port
```

Comment:
Bumping this TODO - uvicorn might have a way to do this already but if not we should be able to bind to port 0 and let the system decide which port to use, so long as we have a way to pass that back to the target here